packages/legend-application/src/stores/storage/WebApplicationClientSideStorageService.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  type PlainObject,
+  SerializationFactory,
+  usingModelSchema,
+} from '@finos/legend-shared';
+import { createModelSchema, list, primitive } from 'serializr';
+import type { GenericLegendApplicationStore } from '../ApplicationStore.js';
+import { WebApplicationClientSideStorage } from './WebApplicationClientSideStorage.js';
+
+export class StorageItem {
+  key!: string;
+  value!: string;
+
+  constructor(key: string, value: string) {
+    this.key = key;
+    this.value = value;
+  }
+
+  static create(key: string, value: string): StorageItem {
+    return new StorageItem(key, value);
+  }
+}
+
+const storageItemModelSchema = createModelSchema(StorageItem, {
+  key: primitive(),
+  value: primitive(),
+});
+
+export class StorageData {
```

Comment:
Not so sure of the complexity here, `StorageService` should have a bunch of `stores` where each `store` is of type `Storage` and the implementation we use right now is `LocalStorage`